### PR TITLE
Pomodoro

### PR DESCRIPTION
**Pull request by emilychangg**
The pomodoro timer feature.

The user can set their own lengths for the work/break intervals or go with the default which is 25 mins work, 5 mins break.

The user can also cancel the timer whenever they want by pressing "c"

### CODE REVIEW

File: src/main/java/console_app/PomodoroController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import main.java.services.pomodoro_running.CancelTimerInput;
+import main.java.data_gateway.PomodoroTimerManager;
+import main.java.services.pomodoro_running.PomodoroObserver;
+import main.java.services.pomodoro_running.PomodoroRunner;
+
+public class PomodoroController {
+    private PomodoroRunner pomodoroRunner;
+    private final PomodoroTimerManager pomodoroTimerManager = new PomodoroTimerManager();
+    private boolean switched = false;
```
Comment (JunruL) [inline src/main/java/console_app/PomodoroController.java:11]: The meaning of `switched` is not clear.
Can you change the attribute name or add documentation for `switched`?

**Review by JunruL — COMMENTED** [review]
Did you create UML for it? I might be easier to review if we have UML.
Note for phase 2: allow user to set the number of Pomodoro rounds.

File: src/main/java/console_app/PomodoroController.java (hunk 1 above)
Comment (jenci2114) [reply src/main/java/console_app/PomodoroController.java:11]: ^^

File: src/main/java/services/pomodoro_running/PomodoroObserver.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        boolean switchNow = false;
+        boolean cancelTimer = false;
+        while(!switchNow && !cancelTimer) {
+            switchNow = pomodoroTimerTask.getSwitchNow();
+            cancelTimer = cancelTimerInput.getCancel();
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException interruptedException) {
+                Thread.currentThread().interrupt();
+            }
```
Comment (jenci2114) [inline src/main/java/services/pomodoro_running/PomodoroObserver.java:28]: maybe add some documentation on this try-except block

File: src/main/java/console_app/ApplicationDriver.java [hunk 3]
```diff
@@ -273,6 +278,47 @@ private static LocalDateTime inputTime() {
...
     }
 
+    /**
+     * Prompts users for their desired work and break intervals
+     * @return the time intervals the user inputted
+     */
+    private static int[] inputPomodoroTime() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Input your desired work and break intervals in (work,break) (in minutes) (no spaces)" +
+                "or 0 for default intervals: ");
```
Comment (JunruL) [inline src/main/java/console_app/ApplicationDriver.java:288]: It does not tell the user what default interval is

File: src/main/java/console_app/ApplicationDriver.java [hunk 4]
```diff
@@ -281,6 +286,47 @@ private static LocalDateTime inputTime() {
...
+            }
+            else {
+                System.out.print("Interval not accepted, please retry.");
+                return inputPomodoroTime();
+            }
+        }
+        else {
+            String[] intervalsSplitString =  intervals.split(",");
+            int[] intervalsSplit = new int[2];
+            for (int i = 0; i < 2; i++){
```
[the remark this answers is not in the document]
Comment (emilychangg) [reply src/main/java/console_app/ApplicationDriver.java]: fixed

**Pull request merged**